one/src/network.rs

```diff
@@ -75,11 +76,17 @@ impl Builder<Init> {
 
 /// Configure the p2p service
 impl Builder<WithStore> {
-    pub async fn with_p2p(
+    pub async fn with_p2p<B>(
         self,
         libp2p_config: Libp2pConfig,
         key_store_path: PathBuf,
-    ) -> anyhow::Result<Builder<WithP2p>> {
+        behvaiour: B,
```

Comment:
```suggestion
        behaviour: B,
```

Reply:
Thanks for the review and spotting the typo. As a quick note, we tried using `request-response` first but found using Behaviour and ConnectionHandler directly to be much easier. Happy to discuss more of our learnings if you'd like.